controller/src/beerocks/master/son_master_thread.cpp

```diff
@@ -608,16 +609,30 @@ bool master_thread::handle_cmdu_1905_autoconfiguration_WSC(Socket *sd,
     * @brief Parse AP-Autoconfiguration WSC which should include one AP Radio Basic Capabilities
     * TLV and one WSC TLV containing M1
     */
-    auto radio_basic_caps = cmdu_rx.addClass<wfa_map::tlvApRadioBasicCapabilities>();
+    static const int num_tlvs                                              = 2;
+    std::shared_ptr<wfa_map::tlvApRadioBasicCapabilities> radio_basic_caps = nullptr;
+    std::shared_ptr<ieee1905_1::tlvWscM1> tlvwscM1                         = nullptr;
+    for (int i = 0; i < num_tlvs; i++) {
+        int type = cmdu_rx.getNextTlvType();
+        if (type == int(wfa_map::eTlvTypeMap::TLV_AP_RADIO_BASIC_CAPABILITIES)) {
+            LOG(DEBUG) << "Found TLV_AP_RADIO_BASIC_CAPABILITIES TLV";
+            radio_basic_caps = cmdu_rx.addClass<wfa_map::tlvApRadioBasicCapabilities>();
+        } else if (type == int(ieee1905_1::eTlvType::TLV_WSC)) {
+            LOG(DEBUG) << "Found TLV_WSC TLV (assuming M1)";
+            tlvwscM1 = cmdu_rx.addClass<ieee1905_1::tlvWscM1>();
+        } else {
+            LOG(ERROR) << "Unexpected TLV " << type;
```

Comment:
To be correct, it should always be allowed to have a vendor-specific TLV.

Reply:
Right, but since we don't support skip TLV yet, and are looking for 2 consecutive TLVs, it is considered unexpected.

Reply:
Changed this a bit - if we see a vendor specific TLV, we assume it is Intel agent (same logic as today)